core/src/main/scala-2.13+/cats/instances/arraySeq.scala

```diff
@@ -0,0 +1,146 @@
+        else fa.flatMap(a => fb.map(b => f(a, b))) // already O(1) if fa is empty
+
+      override def map2Eval[A, B, Z](fa: ArraySeq[A], fb: Eval[ArraySeq[B]])(f: (A, B) => Z): Eval[ArraySeq[Z]] =
+        if (fa.isEmpty) Eval.now(ArraySeq.empty) // no need to evaluate fb
+        else fb.map(fb => map2(fa, fb)(f))
+
+      def foldLeft[A, B](fa: ArraySeq[A], b: B)(f: (B, A) => B): B =
+        fa.foldLeft(b)(f)
+
+      def foldRight[A, B](fa: ArraySeq[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] = {
+        def loop(i: Int): Eval[B] =
+          if (i < fa.length) f(fa(i), Eval.defer(loop(i + 1))) else lb
+
+        Eval.defer(loop(0))
+      }
+
+      override def foldMap[A, B](fa: ArraySeq[A])(f: A => B)(implicit B: Monoid[B]): B =
+        B.combineAll(fa.iterator.map(f))
+
+      def traverse[G[_], A, B](fa: ArraySeq[A])(f: A => G[B])(implicit G: Applicative[G]): G[ArraySeq[B]] =
+        foldRight[A, G[ArraySeq[B]]](fa, Always(G.pure(ArraySeq.untagged.empty))) {
+          case (a, lgvb) =>
+            G.map2Eval(f(a), lgvb)(_ +: _)
+        }.value
+
+      override def mapWithIndex[A, B](fa: ArraySeq[A])(f: (A, Int) => B): ArraySeq[B] =
+        ArraySeq.untagged.tabulate(n = fa.length) { i =>
+          f(fa(i), i)
+        }
+
+      override def zipWithIndex[A](fa: ArraySeq[A]): ArraySeq[(A, Int)] =
+        fa.zipWithIndex
+
+      def tailRecM[A, B](a: A)(fn: A => ArraySeq[Either[A, B]]): ArraySeq[B] =
+        ???
+
+      override def exists[A](fa: ArraySeq[A])(p: A => Boolean): Boolean =
+        fa.exists(p)
+
+      override def forall[A](fa: ArraySeq[A])(p: A => Boolean): Boolean =
+        fa.forall(p)
+
+      override def get[A](fa: ArraySeq[A])(idx: Long): Option[A] =
+        if (idx < fa.length && idx.isValidInt) Some(fa(idx.toInt)) else None
+
+      override def isEmpty[A](fa: ArraySeq[A]): Boolean =
+        fa.isEmpty
+
+      override def foldM[G[_], A, B](fa: ArraySeq[A], z: B)(f: (B, A) => G[B])(implicit G: Monad[G]): G[B] =
+        G.tailRecM((z, 0)) {
+          case (b, i) =>
+            if (i < fa.length) G.map(f(b, fa(i)))(b => Left((b, i + 1)))
+            else G.pure(Right(b))
+        }
+
+      override def fold[A](fa: ArraySeq[A])(implicit A: Monoid[A]): A =
+        A.combineAll(fa)
+
+      override def toList[A](fa: ArraySeq[A]): List[A] =
+        fa.toList
+
+      override def toIterable[A](fa: ArraySeq[A]): Iterable[A] =
+        fa
+
+      override def reduceLeftOption[A](fa: ArraySeq[A])(f: (A, A) => A): Option[A] =
+        fa.reduceLeftOption(f)
+
+      override def find[A](fa: ArraySeq[A])(f: A => Boolean): Option[A] =
+        fa.find(f)
+
+      override def collectFirst[A, B](fa: ArraySeq[A])(pf: PartialFunction[A, B]): Option[B] =
+        fa.collectFirst(pf)
+
+      override def collectFirstSome[A, B](fa: ArraySeq[A])(f: A => Option[B]): Option[B] =
+        fa.collectFirst(Function.unlift(f))
+
+      def functor: Functor[ArraySeq] = this
```

Comment:
What is this needed for?

Reply:
Oh, as I said, I start by copying some exiting instances and adapting what I understood.
I just checked and this is not needed anymore, it probably was required for another typeclass that I am not longer implementing, I will remove it?